src/main/java/com/buddy/api/commons/configurations/security/origin/enums/ClientType.java

```diff
@@ -0,0 +1,19 @@
+package com.buddy.api.commons.configurations.security.origin.enums;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+public enum ClientType {
+    WEB, MOBILE, TOOLS, UNKNOWN;
+
+    public static ClientType fromString(final String type) {
+        return Arrays.stream(ClientType.values())
+            .filter(t -> t.name().equals(type.toLowerCase(Locale.ROOT)))
```

Comment:
aqui acho que teria que ser 
```suggestion
            .filter(t -> t.name().equals(type.toUpperCase(Locale.ROOT)))
```
Imagino que a array fique assim: ["WEB", "MOBILE", "TOOLS", "UNKNOWN"], não?

Reply:
allowed-origins:
    - type: "web"
      code: "550e8400-e29b-41d4-a716-446655440000"
    - type: "mobile"
      code: "123e4567-e89b-12d3-a456-426614174000"
    - type: "tools"
      code: "6ba7b810-9dad-11d1-80b4-00c04fd430c8"
      
  Ele deve comparar com essa properties. O que podia ser feito é deixar igual o enum de fato e remover a chamado ao toUpperCase ou toLowerCase.

Reply:
Mudei para ser tudo upperCase, assim não precisamos mais chamar o método.